src/transceivercontroller/DefaultTransceiverController.ts

```diff
@@ -16,8 +16,35 @@ export default class DefaultTransceiverController implements TransceiverControll
 
   constructor(protected logger: Logger, protected browserBehavior: BrowserBehavior) {}
 
-  setEncodingParameters(_params: Map<string, RTCRtpEncodingParameters>): void {
-    return;
+  async setEncodingParameters(
+    encodingParamMap: Map<string, RTCRtpEncodingParameters>
+  ): Promise<void> {
+    if (!this._localCameraTransceiver || this._localCameraTransceiver.direction !== 'sendrecv') {
+      return;
+    }
+
+    const sender = this._localCameraTransceiver.sender;
+    if (!encodingParamMap || encodingParamMap.size === 0) {
+      return;
+    }
+    const newEncodingParams = Array.from(encodingParamMap.values());
+
+    const oldParam: RTCRtpSendParameters = sender.getParameters();
+    if (!oldParam.encodings) {
+      oldParam.encodings = newEncodingParams;
+    } else {
+      for (let i = 0; i < oldParam.encodings.length; i++) {
+        for (let j = 0; j < newEncodingParams.length; j++) {
+          if (oldParam.encodings[i].rid === newEncodingParams[j].rid) {
+            oldParam.encodings[i].maxBitrate = newEncodingParams[j].maxBitrate;
+            oldParam.encodings[i].scaleResolutionDownBy =
+              newEncodingParams[j].scaleResolutionDownBy;
+          }
+        }
+      }
```

Comment:
I'm not sure I understand this.

You take whatever was already there, and update it with precisely those two parameters.

It looks like this code (a) doesn't allow you to add new encoding parameters after the sender has been created, and (b) doesn't allow you to set new encoding parameters.

Is that deliberate? If so, why?

Reply:
I mainly followed how we did it in Simulcast Transceiver Controller https://github.com/aws/amazon-chime-sdk-js/blob/master/src/transceivercontroller/SimulcastTransceiverController.ts#L47.
I am not exactly sure whether there is a reason we did that but my guess is maybe params can be changed so we cannot just do a direct assignment. Thus, just to be safe, I do it that way.

Reply:
I know you need to mutate the param object given by the sender, and the `rid` is immutable, but that's not really the question — why is this code not something like this:

```typescript
for (const existing of oldParam.encodings) {
  for (const changed of newEncodingParams) {
    if (existing.rid !== changed.rid) {
      continue;
    }
    for (const key in changed) {
      // These can't be changed.
      if (key === 'rid' || key === 'codecPayloadType') {
        continue;
      }
      if (changed.hasOwnProperty(key)) {
        existing[key] = changed[key];
      }
    }
  }
}
```

which would allow the caller to change the `maxFramerate` or other properties?

Reply:
Nice.  Updated the code.